### ISSUE TITLE
AssertReceiveTimeout: resolve module attributes instead of requiring an integer literal

### ISSUE THREAD

**Issue by ck-work00** [Jump-App/credo_checks#37]
`AssertReceiveTimeout` flags any `assert_receive` whose timeout is not an integer literal, so a timeout held in a module attribute counts as a violation even when its value satisfies `min_assert_receive_timeout`.

Example (jump_credo_checks v0.5.0, credo 1.7.19, Elixir 1.20.4):

```elixir
@sms_timeout 5_000

assert_receive {:sms, body}, @sms_timeout
```

With `min_assert_receive_timeout: 2_000` this reports:

```
`assert_receive` timeout must be a literal integer >= 2000, or omitted entirely to use ExUnit's configured default.
```

The value is 5,000, above the floor. The check's `traverse/5` clauses match the timeout argument as an AST node and `maybe_add_assert_issue/6` accepts only an integer, so `{:@, _, [{:sms_timeout, _, _}]}` never passes.

Why it matters: a module attribute is the usual way to name one wait and reuse it across a file. On our suite, enabling the check meant inlining 53 literals across 9 test files, which is the noise the attribute existed to avoid, so we turned the check off.

Suggestions, either would do:
1. Resolve module attributes defined in the same module (`@name value` where `value` is an integer literal) before comparing against the floor, and report when the attribute is not resolvable rather than when it is not a literal.
2. Add a param such as `allow_module_attributes: true` that treats an `@attr` timeout as acceptable, so a team can keep the floor for literals without inlining.

Happy to open a PR for option 1 if you prefer that shape.
